i18n/backends/database/database.go

```diff
@@ -1,13 +1,23 @@
 package database
 
 import (
+	"fmt"
+	"unicode/utf8"
+
 	"github.com/jinzhu/gorm"
 	"github.com/qor/qor/i18n"
 )
 
+/* when using a database with DEFAULT CHARACTER SET utf8mb4 on mysql the table
+creation will fail when not limited in size for key,locale: the compound primary
+key would exceed the size limit:
+Error 1071: Specified key was too long; max key length is 767 bytes
+ideally we'd want to only limit the length to which the index uses the fields,
+but 204/6 should be sufficient for keys and locales.
+*/
 type Translation struct {
-	Key    string `gorm:"primary_key"`
-	Locale string `gorm:"primary_key"`
+	Key    string `gorm:"primary_key" sql:"size:204"`
+	Locale string `gorm:"primary_key" sql:"size:6"`
```

Comment:
will do. what's the max length of fairly commonly used locales?

we also have to change the size of `l10n.Locale`